README.md

```diff
@@ -675,82 +587,315 @@ the client sends a single request and receives multiple responses in a read-only
+const logger = new Logger('Server Test', LogLevel.WARN, [new StreamHandler()]);
+let streamFactory: StreamFactory;
+let middlewareFactory: MiddlewareFactory<any, any, any, any>;
+let idGen: IdGen;
+
+
+
+
+class SquaredNumbers extends ServerHandler<ContainerType, number, number>{
+public handle = async function* (
+input: number,
+):AsyncGenerator<number>{
+for (let i = 0; i<= input; i++){
+yield i*i;
+}
+};
+}
+type Manifest = {
+SquaredNumbers: ServerCaller<number,number>;
+}
+async function startServer() {
+const wss =
+new WebSocket.Server({ port: 1221 });
+const rpcServer =
+await RPCServer.createRPCServer({
+manifest: {
+SquaredNumbers: new SquaredNumbers({}),
+},
+logger,
+idGen,
+});
+
+    wss.on('connection', (ws) => {
+        const { readable, writable } =
+            wsToStream(ws);
+        rpcServer.handleStream({ readable, writable, cancel: () => {} });
+    });
+    return { rpcServer };
+}
+
+async function startClient() {
+return new Promise<RPCClient<Manifest>>((resolve, reject) => {
+const ws = new WebSocket('ws://localhost:1221');
+
+        ws.addEventListener('open', async () => {
+            const { readable, writable } =
+                wsToStream(ws);
+            const rpcClient =
+                await RPCClient.createRPCClient<Manifest>({
+                    manifest: {
+                        SquaredNumbers: new ServerCaller<number, number>(),
+                    },
+                    streamFactory: async () => ({ readable, writable,
+                        cancel: () => {} }),
+                    middlewareFactory,
+                    logger,
+                    idGen,
+                });
+            resolve(rpcClient);
+        });
+
+        ws.addEventListener('error', (err) => {
+            reject(err);
+        });
+    });
+}
+
+function wsToStream(ws: WebSocket): { readable: ReadableStream, writable: WritableStream } {
+let readableController: ReadableStreamDefaultController<any> | null = null;
+
+    const readable = new ReadableStream({
+        start(controller) {
+            readableController = controller;
+        },
+        cancel() {
+            ws.close();
+        },
+    });
+
+    ws.on('message', (chunk: any) => {
+        readableController?.enqueue(chunk);
+    });
+
+    ws.on('close', () => {
+        readableController?.close();
+    });
+
+    const writable = new WritableStream({
+        write(chunk) {
+            ws.send(chunk);
+        },
+        close() {
+            ws.close();
+        },
+        abort() {
+            ws.close();
+        },
+    });
+
+    return { readable, writable };
+}
+
+
+async function execute(rpcClient: RPCClient<Manifest>) {
+try {
+const squaredStream = await rpcClient.methods.SquaredNumbers(235);
+const outputs: Array<number> = [];
+for await(const num of squaredStream) {
+outputs.push(num);
+}
+console.log(`Squared numbers are: ${outputs.join(', ')}`);
+} catch (error) {
+console.error("Error in execute:", error);
+}
+}
```

Comment:
Hey can you make sure code is properly indented with 2 space indent?

Make sure the coding style conforms with our style.

Reply:
In fact even better if you were to put the code into somewhere, and apply the style linting to it to fix it.

Reply:
Then copy onto the `README.md`.